### PR TITLE
[WIP] Vectorization Support for Misaligned Addresses

### PR DESCRIPTION
**Pull request by rdspring1**

### CODE REVIEW

File: test/cpp/jit/test_gpu.cpp [hunk 1]
```diff
@@ -11777,6 +11777,54 @@ TEST(NVFuserTest, FusionVectorization1_CUDA) {
...
+  tv2->axis(0)->parallelize(ParallelType::BIDx);
+  tv2->axis(1)->parallelize(ParallelType::TIDx);
+
+  auto c0 = tv0->cache_after();
+  //auto c2 = tv2->cache_before();
+  c0->computeAt(tv2, -1);
+
+  std::vector<TensorView*> vectorized_tvs = {c0};
+  for (auto tv : vectorized_tvs) {
+    tv->split(-1, 4);
```
Comment (csarofeen) [inline test/cpp/jit/test_gpu.cpp:11804]: I don't think this should be allowed as `c0` was inlined into `tv2` and now you're trying to transform the axis within the compute at. This should throw an error.

File: torch/csrc/jit/codegen/cuda/codegen.cpp [hunk 2]
```diff
@@ -279,34 +279,32 @@ class CudaKernelGenerator : private kir::IrVisitor {
     bool is_vector_op = false;
     size_t vector_word_size = 1;
 
-    if (node->out()->isA<kir::TensorIndex>()) {
-      auto ti = node->out()->as<kir::TensorIndex>();
-      for (auto id : ti->view()->fuserTv()->domain()->domain()) {
-        if (id->getParallelType() != ParallelType::Vectorize) {
-          continue;
-        }
+    if (node->operation() == UnaryOpType::VectorizeSet) {
```
Comment (csarofeen) [inline torch/csrc/jit/codegen/cuda/codegen.cpp:282]: I don't know it makes sense to have a new unary op type to mark this, unless it's guarded from the front end. I wouldn't want this visible to high level APIs as it's a lowering detail. I don't think unary op type is the right way to carry this information.

Comment (csarofeen) [reply torch/csrc/jit/codegen/cuda/codegen.cpp:282]: I would be more comfortable maintaining a for loop stack like we do in other lowering and look at the properties of the for loops.

File: torch/csrc/jit/codegen/cuda/index_compute.cpp [hunk 3]
```diff
@@ -1017,6 +1020,9 @@ kir::TensorIndex* Index::getProducerIndex_impl(
     if (ref_id->getParallelType() == ParallelType::Vectorize) {
       p_id->parallelize(ParallelType::Vectorize);
     }
+    if (ref_id->getParallelType() == ParallelType::VectorizeMisaligned) {
```
Comment (csarofeen) [inline torch/csrc/jit/codegen/cuda/index_compute.cpp:1023]: I can help in indexing compute, but I think we should store and get this information from ForLoop not the tensor view IDs

File: test/cpp/jit/test_gpu.cpp (hunk 1 above)
Comment (rdspring1) [reply test/cpp/jit/test_gpu.cpp:11804]: The computation needs to be inlined with the vectorized read / write when there is a misaligned address.

**Comment by csarofeen** [comment]
@rdspring1 I believe this PR is stale in favor of https://github.com/csarofeen/pytorch/pull/731, if so can you close it?

**Pull request closed without merging**